src/main/java/com/mongodb/hibernate/internal/type/ValueConversions.java

```diff
@@ -166,17 +256,39 @@ private static BigDecimal toDomainValue(BsonDecimal128 value) {
     }
 
     public static String toStringDomainValue(BsonValue value) {
-        return toDomainValue(value.asString());
+        return toDomainValue(value.asString(), String.class);
     }
 
-    private static String toDomainValue(BsonString value) {
-        return value.getValue();
+    /** @see #toBsonValue(char) */
+    @SuppressWarnings("MissingSummary")
+    public static char toCharDomainValue(BsonValue value) {
+        return toDomainValue(value.asString(), Character.class);
+    }
+
+    private static <T> T toDomainValue(BsonString value, Class<T> domainType) {
+        var v = value.getValue();
+        Object result;
+        if (domainType.equals(Character.class)) {
+            result = toDomainValue(v);
+        } else {
+            result = v;
+        }
+        return domainType.cast(result);
+    }
+
+    /** @see #toBsonValue(char) */
+    private static char toDomainValue(String value) {
+        assertTrue(value.length() == 1);
+        return value.charAt(0);
```

Comment:
this method is a little bit strange in the sense that the input is not a BsonValue as in other methods. Also, it is hard to grasp its usage by its usage.

Reply:
Its documentation, or, rather, the method it links to` should explain the situation.

This comment is related to https://github.com/mongodb/mongo-hibernate/pull/95#discussion_r2169285684.